committeeoversightapp/views.py

```diff
@@ -36,78 +45,60 @@ def post(self, request, **kwargs):
-
-            for (field, note) in documents:
-                url = transcript_form.cleaned_data[field]
-                save_document(url, note, event)
-
-            # find and create witnesses
-            for witness in witness_formset.cleaned_data:
-                name = witness.get('name', None)
-                if name:
-                    # add witness as EventParticipant
-                    entity_type = "person"
-                    note = "witness"
-                    new_witness = EventParticipant(
-                                        name=name,
-                                        event=event,
-                                        entity_type=entity_type,
-                                        note=note
-                    )
-                    new_witness.save()
-
-                    #save witness statement urls TK
-                    witness_url = witness.get('url', None)
-                    witness_document = save_document(witness_url, "witness statement", event)
-
-                    #save witness organizations and link to statement urls
-                    organization = witness.get('organization', None)
-                    retired = witness.get('retired', False)
-                    new_witness_details = WitnessDetails(
-                                        witness=new_witness,
-                                        document=witness_document,
-                                        organization=organization,
-                                        retired=retired
-                    )
-                    new_witness_details.save()
+            save_committees(event, committee_form.cleaned_data['name'])
+            save_category(event, category_form.cleaned_data['category'])
+            save_documents(event, transcript_form.cleaned_data)
+            save_witnesses(event, witness_formset.cleaned_data)
 
             # save event source
             source = EventSource(event=event, note="web form")
             source.save()
 
         return redirect('list-event')
 
-class EventList(LoginRequiredMixin, ListView):
-    model = Event
+class EventList(LoginRequiredMixin, TemplateView):
     template_name = 'list.html'
-    context_object_name = 'hearings'
-    queryset = Event.objects.all().order_by('-updated_at')[:500]
+
+class EventListJson(BaseDatatableView):
+    """ Uses django-datatables-view for server-side DataTable processing."""
+    model = Event
+
+    # define the columns that will be returned
+    columns = ['updated_at', 'name', 'start_date', 'id', 'id']
+
+    # define column names that will be used in sorting
+    # order is important and should be same as order of columns
+    # displayed by datatables. For non sortable columns use empty
+    # value like ''
+    order_columns = ['updated_at', 'name', 'start_date', '', '']
+
+    # max limit of records returned
+    max_display_length = 500
```

Comment:
So, we only return 500 records? I know the SOW is ambiguous on this (i.e., 'Improve "View Hearings" page to let users easily browse entered hearings and search by name.')...but I imagined we'd return more elements in the queryset. Am I misunderstanding?

Reply:
This update will display all hearings in the database! This variable refers to max pagination. 

I think I shortened the suggested comment for max_display_length to negative effect. Here's the original, which I'll edit and replace: 

> set max limit of records returned, this is used to protect our site if someone tries to attack our site and make it return huge amount of data

Reply:
Got it! Thanks.